dashboard/app/helpers/aichat_safety_helper.rb

```diff
@@ -0,0 +1,57 @@
+# Provides functionality to detect toxicity in user input and model output used in the AI Chat Lab.
+# Uses various services to check for profanity and toxicity based on DCDO settings.
+module AichatSafetyHelper
+  class ToxicityDetector
+    DEFAULT_TOXICITY_THRESHOLD_USER_INPUT = 0.2
+    DEFAULT_TOXICITY_THRESHOLD_MODEL_OUTPUT = 0.6
+
+    # Checks for toxicity in the given text using various services, determined by DCDO settings.
+    # Returns {text: input (string), blocked_by: serviced that detected toxicity (string), details: filtering details (hash)}
+    def find_toxicity(role, text, locale)
+      if blocklist_enabled?(role)
```

Comment:
I wonder if it would be useful to return a list of words that were blocked from a given input instead of the first found.
WebPurify does this here: https://github.com/code-dot-org/code-dot-org/blob/2051613523466f477eae425e8458137ad53ffbea/lib/cdo/profanity_filter.rb#L40-L41

Reply:
Good question, I'll ask Sam about this

Reply:
Thanks for checking! I see she said it's fine to stop after first blocked word.